### ISSUE TITLE
Results vs Chinchilla

### ISSUE THREAD

**Issue by SpaceCowboy850** [jzhang38/TinyLlama#123]
Do you plan on doing a writeup of your findings?

It would appear that you reached saturation sometime between 2.5T and 3T tokens, but doesn't that blow away the typical ~20x differential that Chinchilla found?

**Comment by jzhang38** [comment]
For this run, two bugs [1](https://whimsical-aphid-86d.notion.site/Release-of-TinyLlama-1-5T-Checkpoints-Postponed-01b266998c1c47f78f5ae1520196d194?pvs=4), [2](https://whimsical-aphid-86d.notion.site/Latest-Updates-from-TinyLlama-Team-7d30c01fff794da28ccc952f327c8d4f?pvs=4) was identified in the course of training so we can not draw a conclusion. We are launching a new training run and will discuss this once that run is finished.

**Comment by Green-Sky** [comment]
@jzhang38 if you are going to fix softmax for your next run ? ( https://www.evanmiller.org/attention-is-off-by-one.html )

> [...] traced the existence of these outlier values to the attention mechanism’s softmax function, [...] finding the off-by-one error [...] .

and corresponding https://github.com/softmax1/Flash-Attention-Softmax-N

**Comment by abodacs** [comment]
@jzhang38 
I heard those tricks make training faster

- Make your vocab size as a multiple of 64 [(Andrej Karpathy says so!)](https://twitter.com/karpathy/status/1621578354024677377)

- Multipack Sampler

The Multipack sampler is designed for padding-free distributed training of large language models

https://github.com/imoneoi/multipack_sampler

**Comment by jzhang38** [comment]
@Green-Sky Yeah thanks for the info and I also read that blog!  My concern over this change is: 1. This is still not a widely adopted practice. 2. It will make TinyLlama incompatible with Llama and thus hard to apply TinyLlama on many other frameworks.

**Comment by Green-Sky** [comment]
@jzhang38 
> 2. It will make TinyLlama incompatible with Llama and thus hard to apply TinyLlama on many other frameworks.

This really depends on how its done. You could alternatively prefix/postfix a frozen 0, which could be written to file (slightly wasteful and needs to be optimized in code).

> 1. This is still not a widely adopted practice.

since you seem to have the funding ...
Why don't you perform a "surgery" as described in the ported flash attention N repository? Also, you stated that you would have more analysis in later papers, this would be ideal for an outlier section :)
